### DISCUSSION TITLE
Using semantic search on a AMD 5600G

### DISCUSSION THREAD

**Discussion by Calimerorulez** [blakeblackshear/frigate#15466] [Ask A Question]
Hi,

I was just experimenting with 0.15-b3, and I'm trying to configure semantic search. Reading the documents, if I want to use hardware acceleration, I need to use the -rocm image.

It's erroring out on the following:

`HIP failure 100: no ROCm-capable device is detected ; GPU=0`

Running in a Proxmox LXC, on Docker, with the GPU of the processor exclusively passed through to the LXC.

**Comment by NickM-27** [comment]
what is your docker compose

**Comment by Calimerorulez** [comment]
The docker compose I used, I used a long time ago. I've added/changed parameters along the way, because I'm running the container quite a while, and I'm updating it as new betas are published.

Do env variables and devices help?


```
HSA_ENABLE_SDMA | 0
LIBAVFORMAT_VERSION_MAJOR | 60
LIBVA_DRIVER_NAME | radeonsi
NVIDIA_DRIVER_CAPABILITIES | compute,video,utility
NVIDIA_VISIBLE_DEVICES | all
OPENCV_FFMPEG_LOGLEVEL | 8
PATH | /usr/lib/btbn-ffmpeg/bin:/usr/local/go2rtc/bin:/usr/local/nginx/sbin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/tempio/bin/
S6_CMD_WAIT_FOR_SERVICES_MAXTIME | 0
S6_LOGGING_SCRIPT | T 1 n0 s10000000 T
TOKENIZERS_PARALLELISM | true
TRANSFORMERS_NO_ADVISORY_WARNINGS | 1
```

![image](https://github.com/user-attachments/assets/2746ccf9-c373-484e-925e-d3a88c846dc4)

Thank you in advance.

**Reply by NickM-27** [reply]
you need to setup the env vars and device mapping according to the docs https://deploy-preview-13787--frigate-docs.netlify.app/configuration/object_detectors#docker-settings-for-overriding-the-gpu-chipset

**Comment by Calimerorulez** [comment]
Thank you, I will try that.

**Comment by Calimerorulez** [comment]
Can I use my GPU alongside my Coral then?

**Reply by NickM-27** [reply]
not for object detection. But you can use coral for object detection and GPU for semantic search

**Comment by Calimerorulez** [comment]
Ok, I just passed through the /dev/kfd device too, using the `-rocm` image, added 

```
  rocm:
    type: rocm
```

to the detector section. Now the error is gone, but indexing now hangs indefinitely, on 0%. The number of todo items in Tracked objects processed is increasing, but the processed ones stay at 0.

When using the 'normal' docker image, and without the rocm detector, indexing is working.

The container restarts itself, I will try to fetch a log.

**Reply by NickM-27** [reply]
There could be some issue, in my testing the 5700G is not very powerful when running rocm so models are quite slow
